### PR TITLE
Thread safing the environment variables on test project and functions refactoring on the API project

### PR DESCRIPTION
**Pull request by Gicehajunior**
Using getenv() is strongly discouraged due to the fact that these functions are not thread safe.

### CODE REVIEW

File: example/index.php [hunk 1]
```diff
@@ -3,8 +3,11 @@
 
 use AfricasTalking\SDK\AfricasTalking;
 
+$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
+$dotenv->load();
+
 $username = "sandbox";
-$apiKey = getenv("API_KEY");
+$apiKey = $_ENV["API_KEY"];
```
Comment (r115) [inline example/index.php:10]: Suggestion;

Could this be named `AFRICASTALKING_API_KEY`? 

`API_KEY` is rather generic and I could be having a similarly named key in my application.

Comment (Gicehajunior) [reply example/index.php:10]: your suggestion is correct.

File: src/AfricasTalking.php [hunk 2]
```diff
@@ -89,6 +82,20 @@ public function __construct($username, $apiKey)
 		]);
 	}
 
+
+	public function base_domain(){
+		define(BASE_DOMAIN, "africastalking.com");
+		define(BASE_SANDBOX_DOMAIN, "sandbox." . self::BASE_DOMAIN);
+
+		if($username === 'sandbox') {
```
Comment (SamuelMwangiW) [inline src/AfricasTalking.php:90]: Hi. I have not tested this so I could entirely be wrong but it seems to me as if `$username` undefined in the function scope. 

I would suggest moving the following lines to above where you're calling the function in `__construct()`

```
$this->username = $username;
```
then change `$username` to `$this->username`

**Comment by calvinkarundu** [comment]
Hey @Gicehajunior ... can you update with the latest changes in the master branch. Also, please update the PR to merge into the develop branch instead.

File: example/.env-example [hunk 3]
```diff
@@ -0,0 +1 @@
+AFRICASTALKING_API_KEY = 235ded9cc0847a38554jnbv74s8fcs5c5dwt5ew521r5ws5r5v524gv
```
Comment (muarachmann) [inline example/.env-example:1]: I don't think this is ok IMO, the docs quite asks you to replace this

Comment (Gicehajunior) [reply example/.env-example:1]: Yeah, I get your case. but I ain't interfering with the logic, However, if you technically look at my logic, we're simply threadsafing the environment variables, in this case, the AFRICASTALKING_API_KEY variable.

Please let me know what need to be changed or so?

**Comment by Gicehajunior** [comment]
> Hey @Gicehajunior ... can you update with the latest changes in the master branch. Also, please update the PR to merge into the develop branch instead.

Done. I hope the changes now suffices?

**Pull request closed without merging**